### PR TITLE
Clean up iscsi target role and adapt it for ubuntu systems

### PR DESCRIPTION
**Pull request by SlothOfAnarchy**

### CODE REVIEW

File: meta/main.yml [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
-  - system
+  - stuvus
   - storage
-  - storage:iscsi
-dependencies:
-  - { role: OndrejHome.targetcli-modules }
+  - iscsi
+  - ubuntu
+  - target
+# dependencies omitted as they can't be installed automatically yet
```
Comment (Scriptkiddi) [inline meta/main.yml]: should install otherwise this role always fails if you dontknow what to do

File: README.md [hunk 2]
```diff
@@ -1,73 +1,63 @@
-targetcli
-=========
+# iscsi-target
 
-Targetcli target iSCSI role which takes care of installing targetcli, enabling it on boot and configuring it based on the provided variable from ansible.
+iSCSI target role that installs and uses targetcli.
 
-NOTE: Modules used by this module were separated into role OndrejHome.targetcli-modules
+## Requirements
```
Comment (Scriptkiddi) [inline README.md]: link to our role?

File: tasks/configure-target.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+---
+- name: define backstore objects for target {{ target.name }}
+  targetcli_backstore:
+    backstore_type: "{{ disk.type }}"
+    backstore_name: "{{ disk.name }}"
+    options: " {{iscsi_disk_path_prefix | default('') }}{{ disk.path }} "
+  with_items:
+    - "{{ target.disks }}"
+  loop_control:
+    loop_var: "disk"
```
Comment (Scriptkiddi) [inline tasks/configure-target.yml]: no quotes

File: tasks/configure-target.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    - save targetcli configuration
+
+- name: define ACLs for iSCSI target {{ target.name }}
+  targetcli_iscsi_acl:
+    wwn: "{{ iscsi_base_wwn }}:{{ target.name }}"
+    initiator_wwn: "{{ initiator }}"
+  with_items:
+    - "{{ target.initiators }}"
+  loop_control:
+    loop_var: "initiator"
```
Comment (Scriptkiddi) [inline tasks/configure-target.yml]: quotes

File: tasks/configure-target.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+- name: assign LUNs to initiators for target {{ target.name }}
+  targetcli_iscsi_lun: 
+    wwn: "{{ iscsi_base_wwn }}:{{ target.name }}"
+    backstore_type: "{{ disk.type }}"
+    backstore_name: "{{ disk.name }}"
+  with_items:
+    - "{{ target.disks }}"
+  loop_control:
+    loop_var: "disk"
```
Comment (Scriptkiddi) [inline tasks/configure-target.yml]: quotes

File: tasks/install.yml [hunk 6]
```diff
@@ -0,0 +1,20 @@
+---
+- name: Install targetcli apt packages
+  apt:
+    name: targetcli
+    state: installed
+    update_cache: yes
```
Comment (Scriptkiddi) [inline tasks/install.yml]: split to own task

File: tasks/install.yml [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  when: ansible_pkg_mgr == "apt"
+
+- name: Install targetcli yum packages
+  yum: 
+    name: targetcli
+    state: installed
+    update_cache: yes
```
Comment (Scriptkiddi) [inline tasks/install.yml]: split to own task

File: meta/main.yml (hunk 1 above)
Comment (SlothOfAnarchy) [inline meta/main.yml]: fixed

File: README.md (hunk 2 above)
Comment (SlothOfAnarchy) [inline README.md]: This role doesn't need any of our roles to work

**Comment by SlothOfAnarchy** [comment]
please rereview

File: README.md (hunk 2 above)
Comment (Scriptkiddi) [inline README.md]: -------------		 +Also, this role is not creating any disks/partitions/LVs. Therefore it is expected that they are already present on machine or created by some other role. 
link to role that creates disks and paritions not in require ments mbut in the description

Comment (SlothOfAnarchy) [inline README.md]: In our case, zfs-storage creates the devices (ZVOLs). This is not included here, as this role is not zfs-specific and can export any device.
You can find the link in the dependencies section where it belongs to.

**Pull request merged**